lasagne/tests/layers/test_special.py

```diff
@@ -446,3 +446,107 @@ def test_prelu(self, init_alpha):
+
+
+class TestRandomizedRectifierLayer:
+    @pytest.fixture
+    def RandomizedRectifierLayer(self):
+        from lasagne.layers.special import RandomizedRectifierLayer
+        return RandomizedRectifierLayer
+
+    def test_high_low(self, RandomizedRectifierLayer):
+        with pytest.raises(ValueError):
+            RandomizedRectifierLayer((None, 3, 28, 28), lower=0.9, upper=0.1)
+
+    def test_nomod_positive(self, RandomizedRectifierLayer):
+        input = np.ones((3, 3, 28, 28)).astype(theano.config.floatX)
+        layer = RandomizedRectifierLayer(input.shape)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out, 1.0)
+
+    def test_low_eq_high(self, RandomizedRectifierLayer):
+        input = np.ones((3, 3, 28, 28)).astype(theano.config.floatX) * -1
+        layer = RandomizedRectifierLayer(input.shape, lower=0.5, upper=0.5)
+        out = layer.get_output_for(input)
+        assert np.allclose(out, -0.5)
+
+    def test_deterministic(self, RandomizedRectifierLayer):
+        input = np.ones((3, 3, 28, 28)).astype(theano.config.floatX) * -1
+        layer = RandomizedRectifierLayer(input.shape, lower=0.4, upper=0.6)
+        out = layer.get_output_for(input, deterministic=True)
+        assert np.allclose(out, -0.5)
+
+    def test_dim_None(self, RandomizedRectifierLayer):
+        import lasagne
+        l_in = lasagne.layers.input.InputLayer((None, 3, 28, 28))
+        layer = RandomizedRectifierLayer(l_in)
+        input = np.ones((3, 3, 28, 28)).astype(theano.config.floatX)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out, 1.0)
+
+    def assert_between(self, layer, input, output):
+        from numpy.testing import assert_approx_equal
+        x = input.flatten()
+        y = output.flatten()
+        for i in range(len(x)):
+            if x[i] >= 0:
+                assert_approx_equal(y[i], x[i])
+            else:
+                assert y[i] <= layer.lower * x[i] and y[i] > layer.upper * x[i]
+
+    def test_get_output_for(self, RandomizedRectifierLayer):
+        input_shape = (3, 3, 28, 28)
+
+        # ensure slope never exceeds [lower,upper)
+        input = np.random.randn(*input_shape).astype(theano.config.floatX)
+        layer = RandomizedRectifierLayer(input_shape, shared_axes=0)
+        self.assert_between(layer, input, layer.get_output_for(input).eval())
+
+        # from here on, we want to check parameter sharing
+        # this is easier to check if the input is all ones
+        input = np.ones(input_shape).astype(theano.config.floatX) * -1
+
+        # default: parameters shared along all but 2nd axis
+        layer = RandomizedRectifierLayer(input_shape)
+        out = layer.get_output_for(input).eval()
+        assert np.allclose(out[0].flatten(), out[-1].flatten())
+        assert not np.allclose(out[0][0].flatten(), out[0][-1].flatten())
+        assert np.allclose(out[0][0][0], out[0][0][-1])
```

Comment:
I think it's easier to read if you just check the variances:
```python
assert [out.var(axis=a) == 0 for a in range(4)] == [False, True, False, False]
```
You can use this scheme for all the tests. The variance should be zero across all axes the randomization is shared over.